blog/2022/05-10-resoto-graph-traversal/index.md

```diff
@@ -0,0 +1,119 @@
+
+```bash
+> search is(digitalocean_vpc) -[0:2]-> | count kind
+// highlight-start
+​digitalocean_app: 1
+​digitalocean_database: 1
+​digitalocean_load_balancer: 2
+​digitalocean_kubernetes_cluster: 2
+​digitalocean_vpc: 2
+​digitalocean_volume: 6
+​digitalocean_droplet: 6
+​total matched: 20
+​total unmatched: 0
+// highlight-end
+```
+
+If we can walk outbound with `-->` and inbound with `<--`, there has to be an option to walk in both directions. The combination of the two approaches is expressed with `<-->`. It is possible to get the list of all reachable resources starting from a specific node. We use the above droplet and ask for all attached resources.
+
+```bash
+> search is(digitalocean_droplet) and id=289061882 <-[0:]-> | count kind
+// highlight-start
+​digitalocean_droplet: 1
+​digitalocean_load_balancer: 1
+​digitalocean_kubernetes_cluster: 1
+​digitalocean_region: 1
+​digitalocean_vpc: 1
+​graph_root: 1
+​cloud: 1
+​digitalocean_image: 1
+​digitalocean_team: 1
+​digitalocean_project: 1
+​digitalocean_volume: 2
+​total matched: 12
+​total unmatched: 0
+// highlight-end
+```
+
+The traversal definition `<-[0:]->` might look strange at first glance. It defines to walk inbound and outbound starting from the current node (`0`). Since the end of the walk is not specified, it will walk until there are no more nodes to traverse. It will walk to the graph root inbound and all possible leafs outbound. The result is then counted by kind. To make this information even more helpful, we can create a graph diagram out of this search using [Graphviz](https://www.graphviz.org/) and the [dot format](/docs/reference/cli/format). The same search as above is used with edges included in the result.
+
+```bash
+> search --with-edges is(digitalocean_droplet) and
+id=289061882 <-[0:]-> | format --dot | write graph.dot
+// highlight-start
+Received a file graph.dot, which is stored to ./graph.dot.
+// highlight-end
+```
+
+We can now use Graphviz to render the graph or use one of the many online tools to visualize the graph. The final result is this graph.
+
+![DigitalOcean Droplet](img/droplet_surrounding.svg)
+
+This diagram reveals a lot of useful information. It shows the related DigitalOcean team and project and the droplets usage as a node in a Kubernetes cluster and backend of a load balancer as well as its attached volumes.
+
+Modeling the resources as graph nodes and the relationships as edges is a natural way to capture this information. You can think of the graph as a digital twin of your infrastructure metadata. Resoto allows you to explore and extract meaningful information from your infrastructure with a simple search.
```

Comment:
"modeling the resources as graph nodes" --> I almost feel like you want to make this as a bold statement at the beginning of the post as well. Come out strong out of the gate.

Reply:
done.